Refactor Homeworks view: Group by subject and add collapsible sections

Louis-htmlcss opened the pull request:
# 🚀 Nouvelle Pull Request

Proposez vos modifications pour améliorer Papillon

## Informations importantes

Merci de vous référer à la documentation sur la contribution si vous avez des questions à propos des pull requests (https://gitbook.getpapillon.xyz/organisation/outils-internes/github)

## Checklist d'avant pull request

Veuillez cocher toutes les cases applicables en remplaçant [ ] par [x].

- [x] Vous avez testé de build le projet avec vos modifications et ce build **a réussi**
- [x] Vous respectez les conventions de codage et de nommage du projet
- [x] Vous utilisez la **tabulation** pour l'indentation afin de maintenir un code lisible
- [x] Cette pull request **n'est pas un duplicata** d'une autre
- [x] Cette pull request est prête à être **revue** (review) et **fusionnée** (merge)
- [x] Il n'y a pas de **`TODO`** (aka des annotations pour du code manquant) dans vos modifications
- [x] Il n'y a pas **d'erreurs de langue** dans votre code (grammaire, vocabulaire, conjugaison, orthographe)
- [x] Les détails des changements ont été décrits ci-dessous
- [x] Cette pull-request n'est pas une **"breaking-change"** (des modifications qui vont entraîner la modification du fonctionnement de certaines fonctionnalités déjà existantes)

## Changelogs proposés

This PR refactors the Homeworks view to improve organization and user experience:

- Removed `subjectData.pretty` from HomeworkItem component
- Grouped homeworks by subject within each day
- Added collapsible sections for each subject
- Implemented a toggle function to expand/collapse subject sections
- Updated the rendering logic to accommodate the new structure
- Added subject icons and homework counts to subject headers
- Adjusted styling for better visual hierarchy

These changes provide a more structured and manageable view of homeworks, allowing users to focus on specific subjects and reducing visual clutter.

Louis-htmlcss commented:
![Screenshot_20240925-161257_Expo Go.png](https://github.com/user-attachments/assets/b44970b2-d746-4aed-a931-56e7754ab936)

Kgeek33 reviewed (approved):
LGTM

godetremy commented:
Pour moi, il semble plus interessant de laisser les devoirs dégroupés, mais en revanche les trié par matière c'est une idée sympathique !

Louis-htmlcss commented:
Ne pas merge, car je dois aussi l'appliquer sur l'écran d'accueil

Louis-htmlcss commented:
> Pour moi, il semble plus interessant de laisser les devoirs dégroupés, mais en revanche les trié par matière c'est une idée sympathique !

Sinon je peux refaire la pr pour rajouter un tri par matière

ecnivtwelve commented on src/views/account/Homeworks/Homeworks.tsx:
```diff
@@ -153,31 +156,47 @@ const WeekView = ({ route, navigation }) => {
+    }, {} as Record<string, Record<string, Homework[]>>);
+
+    const toggleSubject = (day: string, subject: string) => {
+      setCollapsedSubjects(prev => ({
+        ...prev,
+        [`${day}-${subject}`]: !prev[`${day}-${subject}`]
+      }));
+    };
```
Ça serait surement mieux d'avoir les matières **dépliées** par défaut

Louis-htmlcss replied:
Elles sont déjà dépliées par défaut 💀

tom-theret commented:
Très mauvaise idée, c'est hyper pas accessible, puis, je vais pas vous mentir que je ne suis pas très fan du design

tom-theret commented:
Tu regroupe juste tout par matière mais sans emoji ni tiroir toujours tout est visible

Louis-htmlcss commented:
> Tu regroupe juste tout par matière mais sans emoji ni tiroir toujours tout est visible

la je fait un truc pour laisser le truc d'avant mais je rajoute un groupememnt par matieres mais pour toute la semaine

Louis-htmlcss commented:
Ça donne ça maintenant [Screenshot_20240926-211259_Expo Go.png](https://github.com/user-attachments/assets/99d9db47-5631-404b-8718-0182850c00da)

Kgeek33 commented:
> Ça donne ça maintenant [Screenshot_20240926-211259_Expo Go.png](https://github.com/user-attachments/assets/99d9db47-5631-404b-8718-0182850c00da)

J'aime bien mais ça serait bien de voir au-dessus de chaque devoir la date en gris (comme #227 )

Louis-htmlcss commented:
> > Ça donne ça maintenant [Screenshot_20240926-211259_Expo Go.png](https://github.com/user-attachments/assets/99d9db47-5631-404b-8718-0182850c00da)
> 
> J'aime bien mais ça serait bien de voir au-dessus de chaque devoir la date en gris (comme #227 )

Tkt c'est en préparation

tom-theret commented:
> Pour moi, il semble plus interessant de laisser les devoirs dégroupés, mais en revanche les trié par matière c'est une idée sympathique !

Merci

Louis-htmlcss commented:
permet de fix une issue mais jsp laquelle

Louis-htmlcss commented:
This pull request introduces several changes to the homework management feature, focusing on improving the user interface and functionality. The most important changes include adding new props to the `HomeworkItem` component, updating the `WeekView` component to support grouping by subject, and various other enhancements to improve the user experience.

### Enhancements to HomeworkItem Component:
* Added `showSubjectName` and `groupBySubject` props to the `HomeworkItem` component. (`src/views/account/Homeworks/Atoms/Item.tsx`) [[1]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577R13-R15) [[2]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577R38-R50) [[3]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577L67-R86) [[4]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577L85-R117) [[5]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577R131-R152)

### Improvements to WeekView Component:
* Added new state `groupBySubject` and updated the `renderWeek` function to group homeworks by subject when enabled. (`src/views/account/Homeworks/Homeworks.tsx`) [[1]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271R87) [[2]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L145-R179) [[3]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L198-R237)
* Updated the UI to display subject names and counts when grouping by subject is enabled. (`src/views/account/Homeworks/Homeworks.tsx`)

### Code Refactoring and Cleanup:
* Removed redundant comments and improved the readability of the `WeekView` component. (`src/views/account/Homeworks/Homeworks.tsx`) [[1]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L68) [[2]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L129) [[3]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L277) [[4]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L292-L301) [[5]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L551)

### Minor Import Adjustments:
* Added missing imports and optimized existing ones in various files. (`src/views/account/Homeworks/Atoms/Item.tsx`, `src/views/account/Homeworks/Homeworks.tsx`) [[1]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577L1-R1) [[2]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L1-R1) [[3]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L14-R14) [[4]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271R26)

Kgeek33 commented:
> This pull request introduces several changes to the homework management feature, focusing on improving the user interface and functionality. The most important changes include adding new props to the `HomeworkItem` component, updating the `WeekView` component to support grouping by subject, and various other enhancements to improve the user experience.
> 
> ### Enhancements to HomeworkItem Component:
> * Added `showSubjectName` and `groupBySubject` props to the `HomeworkItem` component. (`src/views/account/Homeworks/Atoms/Item.tsx`) [[1]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577R13-R15) [[2]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577R38-R50) [[3]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577L67-R86) [[4]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577L85-R117) [[5]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577R131-R152)
> 
> ### Improvements to WeekView Component:
> * Added new state `groupBySubject` and updated the `renderWeek` function to group homeworks by subject when enabled. (`src/views/account/Homeworks/Homeworks.tsx`) [[1]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271R87) [[2]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L145-R179) [[3]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L198-R237)
> * Updated the UI to display subject names and counts when grouping by subject is enabled. (`src/views/account/Homeworks/Homeworks.tsx`)
> 
> ### Code Refactoring and Cleanup:
> * Removed redundant comments and improved the readability of the `WeekView` component. (`src/views/account/Homeworks/Homeworks.tsx`) [[1]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L68) [[2]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L129) [[3]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L277) [[4]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L292-L301) [[5]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L551)
> 
> ### Minor Import Adjustments:
> * Added missing imports and optimized existing ones in various files. (`src/views/account/Homeworks/Atoms/Item.tsx`, `src/views/account/Homeworks/Homeworks.tsx`) [[1]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577L1-R1) [[2]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L1-R1) [[3]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L14-R14) [[4]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271R26)

J'viens de rentrer chez moi, j'vais tester ça ;)

Kgeek33 commented:
Permet de corriger les issues suivantes : #227 , #193 et #229 

@Louis-htmlcss j'aime bien, c juste dommage le rendu de `En retard` et `Aujourd'hui` mais c'est compréhensible. A la limite, remplacer Aujourd'hui par Ajd et En retard par Avt mais c pas si grave quoi.
![Screenshot_2024-09-27-18-20-09-478_host exp exponent](https://github.com/user-attachments/assets/b059f0a8-4916-46d7-b81c-0521d8187e71)

Louis-htmlcss commented:
> Permet de corriger les issues suivantes : #227 , #193 et #229
> 
> @Louis-htmlcss j'aime bien, c juste dommage le rendu de `En retard` et `Aujourd'hui` mais c'est compréhensible. A la limite, remplacer Aujourd'hui par Ajd et En retard par Avt mais c pas si grave quoi. ![Screenshot_2024-09-27-18-20-09-478_host exp exponent](https://private-user-images.githubusercontent.com/164187100/371608747-b059f0a8-4916-46d7-b81c-0521d8187e71.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0NTQ2NjQsIm5iZiI6MTcyNzQ1NDM2NCwicGF0aCI6Ii8xNjQxODcxMDAvMzcxNjA4NzQ3LWIwNTlmMGE4LTQ5MTYtNDZkNy1iODFjLTA1MjFkODE4N2U3MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyN1QxNjI2MDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNjJkM2VlODg0MWZmZDE2ZWI2M2ZhYjk3NzI2YWI2MDcyYmEwNDdhZjgzMDc0NjQ2ZDFhZDdlMzM4YjZkNGVmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.weLVsRaNzYJfumnMhOw_cIg6fkSu5-8fm2W0Vn6404I)

c'est ou ?

Kgeek33 commented:
> c'est ou ?

Affiche les devoirs de cette semaine (Semaine 4) et tu verras

Louis-htmlcss commented:
> > c'est ou ?
> 
> Affiche les devoirs de cette semaine (Semaine 4) et tu verras

ah oui je vois

Louis-htmlcss commented:
@Kgeek33 j'ai enleve le en retard

Kgeek33 reviewed (approved):
Parfait ça !

cleeemeeent commented:
je review mtn attendez

cleeemeeent reviewed (commented):
Good mais les matières ne sont plus affichées comme avant dans les devoirs, elles sont affichées comme pronote les donne (ex : Français -> FRANCAIS)

cleeemeeent commented:
Je sais pas si c'est pour ça, @Louis-htmlcss si tu peux me dire :))

Louis-htmlcss commented:
> Je sais pas si c'est pour ça, @Louis-htmlcss si tu peux me dire :))

jsp car la j'essaie de fix ca: 
![image](https://github.com/user-attachments/assets/f17c1b56-82ad-4780-b428-7b1614b3bec7)

Kgeek33 commented:
> > Je sais pas si c'est pour ça, @Louis-htmlcss si tu peux me dire :))
> 
> jsp car la j'essaie de fix ca: 
> ![image](https://github.com/user-attachments/assets/f17c1b56-82ad-4780-b428-7b1614b3bec7)
> 

J'ai eu le même problème, je l'ai résolu en faisant la commande suivante (avec les fours d'admin) : `git config --system core.longpaths true`

Louis-htmlcss commented:
> git config --system core.longpaths true

merci tu regale

Louis-htmlcss commented:
> > > Je sais pas si c'est pour ça, @Louis-htmlcss si tu peux me dire :))
> > 
> > 
> > jsp car la j'essaie de fix ca:
> > ![image](https://private-user-images.githubusercontent.com/83782422/371635023-f17c1b56-82ad-4780-b428-7b1614b3bec7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0NjEyNDksIm5iZiI6MTcyNzQ2MDk0OSwicGF0aCI6Ii84Mzc4MjQyMi8zNzE2MzUwMjMtZjE3YzFiNTYtODJhZC00NzgwLWI0MjgtN2IxNjE0YjNiZWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDE4MTU0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5NTNlNWJkYWQ4ZWI3YzY0NzkyNmY2YjY5YjE1M2UwMDkyMDFhMmIyNjJiOWEwMTEwMGJkNGEwMTc2Y2M0YjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DMUICJOPah_0Gn_428caTDW5dtT-hVl6Ofs6U7HW6mg)
> 
> J'ai eu le même problème, je l'ai résolu en faisant la commande suivante (avec les droits d'admin) : `git config --system core.longpaths true`

j'ai ca maietenant, ca fait la meme si je lance le truc demander: 
![image](https://github.com/user-attachments/assets/0ac44135-8dfa-4660-84c8-0e05c5757bae)

Kgeek33 commented:
> > > > Je sais pas si c'est pour ça, @Louis-htmlcss si tu peux me dire :))
> > > 
> > > 
> > > jsp car la j'essaie de fix ca:
> > > ![image](https://private-user-images.githubusercontent.com/83782422/371635023-f17c1b56-82ad-4780-b428-7b1614b3bec7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0NjEyNDksIm5iZiI6MTcyNzQ2MDk0OSwicGF0aCI6Ii84Mzc4MjQyMi8zNzE2MzUwMjMtZjE3YzFiNTYtODJhZC00NzgwLWI0MjgtN2IxNjE0YjNiZWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDE4MTU0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5NTNlNWJkYWQ4ZWI3YzY0NzkyNmY2YjY5YjE1M2UwMDkyMDFhMmIyNjJiOWEwMTEwMGJkNGEwMTc2Y2M0YjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DMUICJOPah_0Gn_428caTDW5dtT-hVl6Ofs6U7HW6mg)
> > 
> > J'ai eu le même problème, je l'ai résolu en faisant la commande suivante (avec les droits d'admin) : `git config --system core.longpaths true`
> 
> j'ai ca maietenant, ca fait la meme si je lance le truc demander: 
> ![image](https://github.com/user-attachments/assets/0ac44135-8dfa-4660-84c8-0e05c5757bae)
> 

Si je comprends bien, t'essayes de faire le `git checkout` en mode administrateur
Relance tes shells en mode normal et réessaie

Louis-htmlcss commented:
> > > > > Je sais pas si c'est pour ça, @Louis-htmlcss si tu peux me dire :))
> > > > 
> > > > 
> > > > jsp car la j'essaie de fix ca:
> > > > ![image](https://private-user-images.githubusercontent.com/83782422/371635023-f17c1b56-82ad-4780-b428-7b1614b3bec7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0NjEyNDksIm5iZiI6MTcyNzQ2MDk0OSwicGF0aCI6Ii84Mzc4MjQyMi8zNzE2MzUwMjMtZjE3YzFiNTYtODJhZC00NzgwLWI0MjgtN2IxNjE0YjNiZWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDE4MTU0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5NTNlNWJkYWQ4ZWI3YzY0NzkyNmY2YjY5YjE1M2UwMDkyMDFhMmIyNjJiOWEwMTEwMGJkNGEwMTc2Y2M0YjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DMUICJOPah_0Gn_428caTDW5dtT-hVl6Ofs6U7HW6mg)
> > > 
> > > 
> > > J'ai eu le même problème, je l'ai résolu en faisant la commande suivante (avec les droits d'admin) : `git config --system core.longpaths true`
> > 
> > 
> > j'ai ca maietenant, ca fait la meme si je lance le truc demander:
> > ![image](https://private-user-images.githubusercontent.com/83782422/371639436-0ac44135-8dfa-4660-84c8-0e05c5757bae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0NjE4MzQsIm5iZiI6MTcyNzQ2MTUzNCwicGF0aCI6Ii84Mzc4MjQyMi8zNzE2Mzk0MzYtMGFjNDQxMzUtOGRmYS00NjYwLTg0YzgtMGUwNWM1NzU3YmFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDE4MjUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcyMjQ0NDQyZWYxMTYzNTcxOWM4MjQ0YmFiNDc1NjZiZmMyNzU1MGFhZmZmYmYxMGU4MzZiMGQ1OGQ4MTRiZGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0ooUQcybKtx8bWYeMqfnGP518mplVVa8dvVwe-fY3T4)
> 
> Si je comprends bien, t'essayes de faire le `git checkout devoir` en mode administrateur Relance tes shells en mode normal et réessaie

je suis pas en admin met j'ai lancer ta coommande en admin

Kgeek33 commented:
> > > > > > Je sais pas si c'est pour ça, @Louis-htmlcss si tu peux me dire :))
> > > > > 
> > > > > 
> > > > > jsp car la j'essaie de fix ca:
> > > > > ![image](https://private-user-images.githubusercontent.com/83782422/371635023-f17c1b56-82ad-4780-b428-7b1614b3bec7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0NjEyNDksIm5iZiI6MTcyNzQ2MDk0OSwicGF0aCI6Ii84Mzc4MjQyMi8zNzE2MzUwMjMtZjE3YzFiNTYtODJhZC00NzgwLWI0MjgtN2IxNjE0YjNiZWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDE4MTU0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5NTNlNWJkYWQ4ZWI3YzY0NzkyNmY2YjY5YjE1M2UwMDkyMDFhMmIyNjJiOWEwMTEwMGJkNGEwMTc2Y2M0YjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DMUICJOPah_0Gn_428caTDW5dtT-hVl6Ofs6U7HW6mg)
> > > > 
> > > > 
> > > > J'ai eu le même problème, je l'ai résolu en faisant la commande suivante (avec les droits d'admin) : `git config --system core.longpaths true`
> > > 
> > > 
> > > j'ai ca maietenant, ca fait la meme si je lance le truc demander:
> > > ![image](https://private-user-images.githubusercontent.com/83782422/371639436-0ac44135-8dfa-4660-84c8-0e05c5757bae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0NjE4MzQsIm5iZiI6MTcyNzQ2MTUzNCwicGF0aCI6Ii84Mzc4MjQyMi8zNzE2Mzk0MzYtMGFjNDQxMzUtOGRmYS00NjYwLTg0YzgtMGUwNWM1NzU3YmFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDE4MjUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcyMjQ0NDQyZWYxMTYzNTcxOWM4MjQ0YmFiNDc1NjZiZmMyNzU1MGFhZmZmYmYxMGU4MzZiMGQ1OGQ4MTRiZGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0ooUQcybKtx8bWYeMqfnGP518mplVVa8dvVwe-fY3T4)
> > 
> > Si je comprends bien, t'essayes de faire le `git checkout devoir` en mode administrateur Relance tes shells en mode normal et réessaie
> 
> je suis pas en admin met j'ai lancer ta coommande en admin

T'as fais `git clone` en mode admin ?
Si oui, supprime le repos en local et refais `git clone` normalement

cleeemeeent commented:
> permet de fix une issue mais jsp laquelle

#229 je pense? 
j'ai plus le crash quand on clique sur un cours de l'EDT

Kgeek33 commented:
> > permet de fix une issue mais jsp laquelle
> 
> #229 je pense? 
> j'ai plus le crash quand on clique sur un cours de l'EDT

Oui, c une des 3 issues corrigées (j'ai listé toutes les issues au tout 1er msg)

Louis-htmlcss commented:
> > > > > > > Je sais pas si c'est pour ça, @Louis-htmlcss si tu peux me dire :))
> > > > > > 
> > > > > > 
> > > > > > jsp car la j'essaie de fix ca:
> > > > > > ![image](https://private-user-images.githubusercontent.com/83782422/371635023-f17c1b56-82ad-4780-b428-7b1614b3bec7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0NjEyNDksIm5iZiI6MTcyNzQ2MDk0OSwicGF0aCI6Ii84Mzc4MjQyMi8zNzE2MzUwMjMtZjE3YzFiNTYtODJhZC00NzgwLWI0MjgtN2IxNjE0YjNiZWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDE4MTU0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5NTNlNWJkYWQ4ZWI3YzY0NzkyNmY2YjY5YjE1M2UwMDkyMDFhMmIyNjJiOWEwMTEwMGJkNGEwMTc2Y2M0YjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DMUICJOPah_0Gn_428caTDW5dtT-hVl6Ofs6U7HW6mg)
> > > > > 
> > > > > 
> > > > > J'ai eu le même problème, je l'ai résolu en faisant la commande suivante (avec les droits d'admin) : `git config --system core.longpaths true`
> > > > 
> > > > 
> > > > j'ai ca maietenant, ca fait la meme si je lance le truc demander:
> > > > ![image](https://private-user-images.githubusercontent.com/83782422/371639436-0ac44135-8dfa-4660-84c8-0e05c5757bae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0NjE4MzQsIm5iZiI6MTcyNzQ2MTUzNCwicGF0aCI6Ii84Mzc4MjQyMi8zNzE2Mzk0MzYtMGFjNDQxMzUtOGRmYS00NjYwLTg0YzgtMGUwNWM1NzU3YmFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDE4MjUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcyMjQ0NDQyZWYxMTYzNTcxOWM4MjQ0YmFiNDc1NjZiZmMyNzU1MGFhZmZmYmYxMGU4MzZiMGQ1OGQ4MTRiZGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0ooUQcybKtx8bWYeMqfnGP518mplVVa8dvVwe-fY3T4)
> > > 
> > > 
> > > Si je comprends bien, t'essayes de faire le `git checkout devoir` en mode administrateur Relance tes shells en mode normal et réessaie
> > 
> > 
> > je suis pas en admin met j'ai lancer ta coommande en admin
> 
> T'as fais `git clone` en mode admin ? Si oui, supprime le repo en local et refais `git clone` normalement

c'est bon merci beaucoup

Louis-htmlcss commented:
@ecnivtwelve

Louis-htmlcss commented:
This pull request introduces several changes to the `Homeworks` feature to enhance functionality and improve the user experience. The key updates include the addition of new properties to the `HomeworkItem` component, the use of `useMemo` for performance optimization, and the introduction of a new grouping mechanism for displaying homework items.

### Enhancements to Homework Display:

* [`src/views/account/Home/Elements/HomeworksElement.tsx`](diffhunk://#diff-04b22e0620652ecdc74ee0b824a55297463cab63f68c7ec5a7ca6f7d59581c91R62-R63): Added `showSubjectName` and `groupBySubject` properties to the `HomeworkItem` component to allow better organization and display of homework items. [[1]](diffhunk://#diff-04b22e0620652ecdc74ee0b824a55297463cab63f68c7ec5a7ca6f7d59581c91R62-R63) [[2]](diffhunk://#diff-04b22e0620652ecdc74ee0b824a55297463cab63f68c7ec5a7ca6f7d59581c91R76-R77)
* [`src/views/account/Homeworks/Atoms/Item.tsx`](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577L1-R1): Introduced the `useMemo` hook to calculate the remaining days for each homework item and added new UI elements to display the subject name and group by subject. [[1]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577L1-R1) [[2]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577R13-R15) [[3]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577R38-R49) [[4]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577L67-R85) [[5]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577L85-R116) [[6]](diffhunk://#diff-a2bb7e58ab074a931e9d6c97889692c37ac352c4ee6926805479ae733ac7e577R130-R152)

### Grouping and Filtering Improvements:

* [`src/views/account/Homeworks/Homeworks.tsx`](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L1-R1): Implemented a new grouping mechanism that allows homework items to be grouped by subject or date, with additional filtering options based on search terms and completion status. [[1]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L1-R1) [[2]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L14-R14) [[3]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271R26-R27) [[4]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L38-R53) [[5]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L68) [[6]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271R97) [[7]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L129) [[8]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L145-R200) [[9]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L198-R281) [[10]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L257-R324) [[11]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L277) [[12]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L292-L301) [[13]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271R553-R602) [[14]](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L551)

### Code Optimization:

* [`src/views/account/Homeworks/Homeworks.tsx`](diffhunk://#diff-9e66512c735f50f235baba2cb8215a63cbf5182e38ae7a0bd062c8f4edd78271L145-R200): Refactored the `renderWeek` function to use `useCallback` for better performance and readability.

These changes collectively enhance the user interface and functionality of the `Homeworks` feature, making it more intuitive and efficient for users to manage their homework assignments.

Kgeek33 commented:
Quand les devs sont regroupés par date, peux-tu mettre `demain` --> `Demain` (_ un détail je sais 🤣_)

Kgeek33 reviewed (approved):
LGTM 👍

Louis-htmlcss commented:
![Screenshot_20241014-172716_Expo Go.png](https://github.com/user-attachments/assets/b55eddb5-b91d-4c16-bf04-3498c121ebaa)

![Screenshot_20241014-172707_Expo Go.png](https://github.com/user-attachments/assets/5bd0093d-6677-4514-b7ed-80b2676c7880)

![Screenshot_20241014-172701_Expo Go.png](https://github.com/user-attachments/assets/13434dbb-ec44-4d3f-a0e1-57301d9e7fe5)

@tryon-dev

tryon-dev commented:
![image](https://github.com/user-attachments/assets/f7b004fe-f9d3-4248-87c8-4eea7b487934)
c'est quoi les -16h ?

Kgeek33 commented:
Ça, ça arrive quand le devoir était pour la veille j'crois

Louis-htmlcss commented:
> Ça, ça arrive quand le devoir était pour la veille j'crois

Yes

Kgeek33 commented:
Ah t'as des conflits 😢

Kgeek33 commented:
This PR will closed #227

Kgeek33 commented:
Raté, je reprends

Closed #227

Kgeek33 commented:
ah ben marche pas, @Louis-htmlcss tu l'ajoutes dans ta description de ta pr pour que ça closed automatiquement ?

Louis-htmlcss commented:
> ah ben marche pas, @Louis-htmlcss tu l'ajoutes dans ta description de ta pr pour que ça closed automatiquement ?

Comment on fait ?

Louis-htmlcss commented:
close #227

Louis-htmlcss commented:
close: #227

Kgeek33 commented:
> close: #227

Marche pas, tant pis pg

Kgeek33 commented:
> @Kgeek33 juste une question, t'es sur le discord ?

Non j'y suis pas

Louis-htmlcss commented:
> > close: #227
> 
> Marche pas, tant pis pg

J'ai réussi c'est bon

Louis-htmlcss commented:
> > @Kgeek33 juste une question, t'es sur le discord ?
> 
> Non j'y suis pas

Bas tu pourrais le rejoindre 😄

Kgeek33 commented:
> > > @Kgeek33 juste une question, t'es sur le discord ?
> > 
> > Non j'y suis pas
> 
> Bas tu pourrais le rejoindre 😄

Je ferai pdnt les vacances, merci de proposer 🙂

Theoyeah commented:
c même pas encrypté contre l'ia :broken_heart:
